assets/js/payment-method-extensions/payment-methods/stripe/apple-pay/apple-pay-express.js

```diff
@@ -223,17 +221,33 @@ const ApplePayExpressComponent = ( {
 		}
 	};
 
+	const onPaymentProcessing = () => {
+		const handlers = eventHandlers.current;
+		if (
+			typeof handlers.sourceEvent === 'function' &&
+			applePayProcessing
+		) {
+			return {
+				billingData: getBillingData( handlers.sourceEvent ),
+				paymentMethodData: getPaymentMethodData(
+					handlers.sourceEvent,
+					PAYMENT_METHOD_NAME
+				),
+			};
+		}
+		return true;
+	};
+
 	const onCheckoutComplete = ( forSuccess = true ) => () => {
 		const handlers = eventHandlers.current;
 		if (
 			typeof handlers.sourceEvent === 'function' &&
-			currentPaymentStatus.current.currentStatus.isSuccessful
+			applePayProcessing
```

Comment:
I'm a bit confused, shouldn't it be:
```suggestion
			! applePayProcessing
```

Reply:
@Aljullu It looks like `applePayProcessing` might be to track if the button to process applepay was pressed, so that this logic only runs on applepay requests? Docblocks would help clear this up perhaps.

Reply:
Mike is correct here. Naming it `applePayIsProcessing` might also help.   What may be confusing is that originally payment methods took care of updating status and at this point, the payment method would have been set to successful. But now payment method status is set by checkout and internally for the payment method at this point the payment method just cares whether it is in an internal processing state (i.e. the button has been clicked and the browser payment window hasn't closed yet).

Reply:
I'm going to leave this as is for this pull. I have an in progress pull (just waiting for other pulls to get merge before I wind up) that changes apple-pay to payment-request everywhere because with a few tweaks I was able to just support the payment request api in it's entirety (so includes apple pay, google pay, microsoft pay and any other browsers using the payment request api). So there's some naming changes etc happening in that pull.